Website

reubenharry opened the pull request:
Big PR, bumping the version to 1.0.0. Hosts monad-bayes website and docs, updates the notebooks, fixes smaller issues.

Commit:
analytic

Commit:
docs

Commit:
docs

Commit:
docs

Commit:
docs

reubenharry committed:
inference

reubenharry committed:
executable docs

reubenharry committed:
more comments, added BayesianModel

reubenharry committed:
Remove MaybeT and use ExceptT

reubenharry committed:
remove vscode and update gitignore

reubenharry committed:
delete additional changes: Failure

reubenharry committed:
delete broken files

reubenharry committed:
update test for enumerator

reubenharry committed:
update models

reubenharry committed:
tips on inference

reubenharry committed:
density

reubenharry committed:
analytic

reubenharry committed:
analytic

reubenharry committed:
more pipes

reubenharry committed:
update docs

reubenharry committed:
linear outliers as in Gen tutorial

reubenharry committed:
update docs

reubenharry committed:
cleanup SMC.hs

reubenharry committed:
more cleanup, and explicit imports

reubenharry committed:
remove Rejection

reubenharry committed:
cleanup

reubenharry committed:
enumerate

reubenharry committed:
binning

reubenharry committed:
comment

reubenharry committed:
explore

reubenharry committed:
nested inference example

reubenharry committed:
found bug in Gamma

reubenharry committed:
debugging

reubenharry committed:
port lazyppl monad into monad-bayes

reubenharry committed:
mh

reubenharry committed:
add conjugacy tests

reubenharry committed:
working branch

reubenharry committed:
tried symbolic

reubenharry committed:
no symbolic

reubenharry committed:
Merge branch 'analytic' into models

reubenharry committed:
remove reflection

reubenharry committed:
Merge branch 'analytic' into models

reubenharry committed:
midway: need to finish hmm and enumerator:

reubenharry committed:
update docs

reubenharry committed:
DP mixture: remove

reubenharry committed:
build docs manually

reubenharry committed:
insegel theme

reubenharry committed:
stanford theme

reubenharry committed:
lint

reubenharry committed:
lint

Commit:
ormulo

Commit:
nix ormolu fixes:
git push

reubenharry committed:
merge main

reubenharry committed:
docs

reubenharry committed:
trying to fix mathjax

reubenharry committed:
Merge branch 'master' into docs

reubenharry committed:
merge docs

reubenharry committed:
remove empirical

reubenharry committed:
merge docs

reubenharry committed:
notebooks

reubenharry committed:
Merge branch 'notebook' into histogram

reubenharry committed:
notebooks: sampling

reubenharry committed:
merge cleanup

reubenharry committed:
merge notebook

reubenharry committed:
cleaning up notebooks

reubenharry committed:
remove helpers

reubenharry committed:
Merge branch 'cleanup' into notebook

reubenharry committed:
when

reubenharry committed:
no helpers

reubenharry committed:
models: lts

reubenharry committed:
models: lts

reubenharry committed:
SMC notebook

reubenharry committed:
merged

reubenharry committed:
weighted sampling

reubenharry committed:
formatting

reubenharry committed:
Merge branch 'notebook' into cleanup

reubenharry committed:
working on analytic

reubenharry committed:
test enumerator

reubenharry committed:
discovered a bug in Integrator

reubenharry committed:
first attempt to fix normalization bug

reubenharry committed:
bug seems to be fixed

reubenharry committed:
update tests

reubenharry committed:
working tests: normalNormal goes awry if observations are huge

reubenharry committed:
cleanup

reubenharry committed:
Bayesian in class, and remove LinearWithOutliers

reubenharry committed:
model updates

Commit:
yaml

Commit:
prepare for PR

Commit:
address warnings

Commit:
review

Commit:
more warnings fixed

reubenharry committed:
merged

Commit:
less warnings

reubenharry committed:
update ghc version stated in cabal

reubenharry committed:
Merge branch 'cleanup' of github.com:tweag/monad-bayes into cleanup

Commit:
remove warnings

Commit:
merge main

Commit:
fix build

Commit:
version bump

Commit:
merged but tests fail

Commit:
remove safe

Commit:
Merge branch 'cleanup' into histogram

Commit:
fix merge

reubenharry committed:
fix rmsmc bug you introduced

reubenharry committed:
integrator as MonadInfer

reubenharry committed:
updates

reubenharry committed:
updates

reubenharry committed:
updates

reubenharry committed:
fix the integrator

reubenharry committed:
fix the normalization of the integrator

reubenharry committed:
merged

reubenharry committed:
inference

reubenharry committed:
rope in analyticT

Commit:
remove integratorT

Commit:
nix notebooks

Commit:
notebooks

Commit:
rmsmc

Commit:
lazy ppl

Commit:
hoistFirst bug again

Commit:
hoistFirst bug again

Commit:
diagrams

Commit:
fp notebook

Commit:
remove integratorT

Commit:
Merge branch 'cleanup' into histogram

Commit:
Merge branch 'histogram' into analytic

Commit:
changelog and version bump

Commit:
Merge branch 'models' into lazyppl

Commit:
presentation

Commit:
notebooks

Commit:
notebooks

Commit:
cleanup

Commit:
docs

Commit:
docs

Commit:
presentation

Commit:
update talk

Commit:
notebook

Commit:
works, but readd basic

Commit:
first working example of marginalization

Commit:
renaming

Commit:
continue cleanup

Commit:
tests: not yet integrated in

Commit:
tests: integrated in

Commit:
bayesian pmmh

Commit:
switch over to new smc

Commit:
switch over to new smc

Commit:
improve resampleGeneric sum

Commit:
independent in Class

reubenharry committed:
parser

reubenharry committed:
nbs

reubenharry committed:
no numSteps

reubenharry committed:
make clean

reubenharry committed:
merge

reubenharry committed:
fix tests

reubenharry committed:
fix tests

reubenharry committed:
fix tests

reubenharry committed:
fix tests

reubenharry committed:
pmmh

reubenharry committed:
withParticles

reubenharry committed:
withParticles

reubenharry committed:
Sequential.Free

reubenharry committed:
Merge branch 'api' into sequentialCata

reubenharry committed:
Sequential.Free

reubenharry committed:
Density.Free + docs

reubenharry committed:
docs

reubenharry committed:
docs

reubenharry committed:
Sampler.Strict

reubenharry committed:
update nix

reubenharry committed:
cabal format

reubenharry committed:
update nix

reubenharry committed:
fix build

reubenharry committed:
merge sampler changes, fix changelog

reubenharry committed:
bump changelog and version. bump ghc version to 9.2.3

reubenharry committed:
update docs and change mcmc so unweighted is not needed

reubenharry committed:
remove sequentialCata

reubenharry committed:
Merge branch 'simplifyFreeSampler' into lazyppl

reubenharry committed:
nbs

reubenharry committed:
nbs

reubenharry committed:
add docs and Distribution, Measure, Kernel

reubenharry committed:
merge

reubenharry committed:
fix cabal

reubenharry committed:
merge

reubenharry committed:
Merge branch 'lazyppl' into notebooks

reubenharry committed:
updating notebooks

reubenharry committed:
Merge branch 'lazyppl' into notebooks

reubenharry committed:
Merge branch 'distribution' into notebooks

reubenharry committed:
updated notebooks

reubenharry committed:
mhTrans'

reubenharry committed:
fix mh

reubenharry committed:
Merge branch 'simplifyFreeSampler' into lazyppl

reubenharry committed:
merge

reubenharry committed:
website

reubenharry committed:
website

reubenharry committed:
bayesian

reubenharry committed:
fix import

reubenharry committed:
fix pmmh example

reubenharry committed:
tui

reubenharry committed:
no streamly

reubenharry committed:
lint

reubenharry committed:
lint

reubenharry committed:
piped mcmc

reubenharry committed:
merge

reubenharry committed:
merge

aspiwack commented on README.md:
```diff
@@ -11,12 +13,11 @@
 <!-- See the [documentation](https://monad-bayes.netlify.app/) for a quick-start user guide and a reference overview of how it all works. -->
 
-Created by [Adam Scibior][adam-web] ([@adscib][adam-github]), documentation, website and various extras by [Reuben][reuben-web], maintained by [Tweag I/O][tweagio].
+Created by [Adam Scibior][adam-web] ([@adscib][adam-github]), documentation, website and newer features by [Reuben][reuben-web], maintained by [Tweag I/O][tweagio].
```
```suggestion
Created by [Adam Scibior][adam-web] ([@adscib][adam-github]), documentation, website and newer features by [Reuben][reuben-web], maintained by [Tweag][tweagio].
```

reubenharry replied:
Here is the current draft of the website: https://monad-bayes-site.netlify.app/_site/about.html I am currently using my own netlify to host it. I would definitely prefer to switch this to the Tweag Netlify, but I'll need access to do that. I'm happy with any url.

aspiwack replied:
@garbas please advise.

garbas replied:
Sorry for late reply.

We can of course move this to Tweag's Netlify, but I would encourage you to look at Cloudflare Pages (this is what we are using for tweag.io) since it provides much better services.

@reubenharry Please set up a call and lets get to access to either Netlify or Cloudflare.

aspiwack replied:
How do we provide hosting for https://nickel-lang.org/, for instance? This is a similar situation.

reubenharry replied:
I've finished my fellowship at Tweag, so while I'm happy to hand over the page to Tweag's cloudflare or netlify, I don't have much time to work out the logistics. My suggestion is we put this aside for the current PR.

aspiwack reviewed (commented):
Just checking for due diligence:
- What will be the website's url?
  - Does the infra team own the domain name?
- Are we using Tweag's Netlify account for hosting?

reubenharry committed:
fixing nix

reubenharry committed:
notebooks

reubenharry committed:
notebooks

reubenharry committed:
docs

reubenharry committed:
Merge branch 'master' into notebooks

reubenharry committed:
fix tui

mknorps commented:
Just having a fist look - is there a particular reason why you kept `.ipynb_checkpoints` directories? I usually added  them to `.gitignore` as their content change a lot and no comparison with previous version is needed.

idontgetoutmuch commented:
> Just having a fist look - is there a particular reason why you kept `.ipynb_checkpoints` directories? I usually added them to `.gitignore` as their content change a lot and no comparison with previous version is needed.

Great suggestion

dependabot[bot] committed:
chore(deps): bump DeterminateSystems/update-flake-lock from 13 to 14 (#193)

Bumps [DeterminateSystems/update-flake-lock](https://github.com/DeterminateSystems/update-flake-lock) from 13 to 14.
- [Release notes](https://github.com/DeterminateSystems/update-flake-lock/releases)
- [Commits](https://github.com/DeterminateSystems/update-flake-lock/compare/v13...v14)

---
updated-dependencies:
- dependency-name: DeterminateSystems/update-flake-lock
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

reubenharry committed:
new notebook

reubenharry committed:
new notebook

mknorps commented on README.md:
```diff
@@ -11,12 +13,11 @@
 
-Now that `monad-bayes` has been released on Hackage, we will focus on improving
-documentation, adding a variety of applications, improving the API, and making inference more easily customizable.
+Now that `monad-bayes` has been released on Hackage, and the documentation and API has been updated, we will focus on adding new features. See the issues to get a sense, and please feel free to make requests.
```
```suggestion
Now that `monad-bayes` has been released on Hackage, and the documentation and the API has been updated, we will focus on adding new features. See the Github issues to get a sense of what is being prepared, and please feel free to make requests.
```

mknorps commented on README.md:
```diff
@@ -48,12 +49,14 @@ for probabilistic programs][thesis-doi]. Thesis. University of Cambridge.
 
 2. This should open a shell, from which you can run `jupyter-lab` to load the notebooks
 
+Your mileage may vary. 
```
I don't get it, could you explain?

reubenharry replied:
It's an English idiom. Roughly means: "I don't guarantee that this will work for you". We can change the wording, but I just wanted to let people know that they might need to do some troubleshooting to get it to work for them.

mknorps replied:
Ah, I see now :) Nice idiom indeed!
On the other hand it would be good to take the setup pain away from a reader's shoulders - in which Nix should help.
I suggest a small reformulation - instead of :


> **To view the notebooks, go to the website**. To use the notebooks interactively, you will first need `nix`. Then:
> 
> 1. Run `nix develop --system x86_64-darwin --extra-experimental-features nix-command --extra-experimental-features flakes`
> 
> 2. This should open a shell, from which you can run `jupyter-lab` to load the notebooks
> 
> Your mileage may vary. 
>

Write:

**To view the notebooks, go to the website**. To use the notebooks interactively:
1. Compile the source: `stack build`
2. If you do not have `nix` [install it](https://nixos.org/download.html).
3. Run `nix develop --system x86_64-darwin --extra-experimental-features nix-command --extra-experimental-features flakes` - this should open a nix shell.
4. Run `jupyter-lab` from the nix shell to load the notebooks

Your mileage may vary. 


It is pretty much the same but keeps the prerequisite to run notebooks in one place.

reubenharry replied:
I like this, let's do it.

mknorps replied:
done in https://github.com/tweag/monad-bayes/pull/179/commits/2ef32d9758ce09a48439eed39ef349f7645eb2ac

mknorps commented on monad-bayes-site/about.md:
```diff
@@ -6,19 +6,30 @@ Monad-Bayes is a library for **probabilistic programming** written in **Haskell*
 **Define distributions** [as programs](../Introduction.html)
 
-**Perform inference** [with a variety of methods](/tutorials.html) 
+**Perform inference** [with a variety of standard methods, which you can extend](/tutorials.html) 
```
```suggestion
**Perform inference** [with a variety of standard methods](/tutorials.html) 
```
I liked previous formulation better - it was concise.

mknorps commented on monad-bayes-site/about.md:
```diff
@@ -6,19 +6,30 @@ Monad-Bayes is a library for **probabilistic programming** written in **Haskell*
-**Perform inference** [with a variety of methods](/tutorials.html) 
+**Perform inference** [with a variety of standard methods, which you can extend](/tutorials.html) 
 
+**Integrate with arbitrary Haskell code** [like this](/examples.html) because Monad-Bayes is just a library, not a separate language
```
```suggestion
**Integrate with Haskell code** [like this](/examples.html) because Monad-Bayes is just a library, not a separate language
```
I think those main advantages of the monad-bayes should be presented in a short form and elaborated later if needed.

mknorps commented on monad-bayes-site/about.md:
```diff
@@ -6,19 +6,30 @@ Monad-Bayes is a library for **probabilistic programming** written in **Haskell*
+     width="720" 
+     height="300" />
 
+`model` is a mixture of Gaussians, defined as a program. Its type `Distribution Double` shows that it is a distribution over reals. `image` is a program too: as its type shows, it is a distribution over plots. In particular, plots that arise from forming a 200 bin histogram out of 100000 iid draws from `model`. To sample from `image`, we simply write `sampler image`, with the result shown below:
```
```suggestion
The program `model` is a mixture of Gaussians. Its type `Distribution Double` represents a distribution over reals. 
`image` is a program too: as its type shows, it is a distribution over plots. In particular, plots that arise from forming a 200 bin histogram out of 100000 independent identically distributed (iid) draws from `model`. 
To sample from `image`, we simply write `sampler image`, with the result shown below:
```
Please use new lines for new sentences - it will be easier to read.

reubenharry replied:
Minor edit:

"The program `model` is a mixture of Gaussians. Its type `Distribution Double` represents a distribution over reals. "

mknorps replied:
ok

mknorps commented on monad-bayes-site/about.md:
```diff
@@ -6,19 +6,30 @@ Monad-Bayes is a library for **probabilistic programming** written in **Haskell*
+# Example
 
 
+<img src="../images/code_example.png" 
```
If we could use a regular code instead of a picture it would be better - a reader could copy it if needed and it the site will be more coherent.

reubenharry replied:
Good point, I agree. Some other feedback on the website was given here: https://discourse.haskell.org/t/probabilistic-programming-in-haskell/4867/4

mknorps replied:
Addressed in https://github.com/tweag/monad-bayes/pull/201

mknorps commented on monad-bayes-site/_site/about.html:
```diff
@@ -23,31 +23,40 @@
             <p>Monad-Bayes is a library for <strong>probabilistic programming</strong> written in <strong>Haskell</strong>.</p>
-<p><strong>Define distributions</strong> <a href>as programs</a></p>
-<p><strong>Perform inference</strong> <a href>with a variety of methods</a></p>
+<p><strong>Define distributions</strong> <a href="../Introduction.html">as programs</a></p>
```
Is this automatically generated from about.md?

reubenharry replied:
The html notebook? It is not.

mknorps replied:
I checked and the `about.html` site is generated from `about.md` by running in `monad-bayes-site` program `site.hs`.

I do not see such link between notebook tutorials and examples and their html counterparts.

mknorps replied:
If there is no such automatization I propose making few issues of it:
1. Check and remove notebooks from under the `notebooks` directory which are currently duplicated in subdirectories (like `examples`)
2. Add a way to build html pages from notebooks and pandoc documentation as a script (Makefile?) and a Readme to `monad-bayes-site`

reubenharry replied:
There is not currently automation (because I didn't know how). pandoc seems like the way, and that seems like a sensible approach.

mknorps replied:
It will be solved in #210

mknorps commented on docs/source/probprog.md:
```diff
@@ -18,39 +18,38 @@ y_{n}=\alpha+\beta x_{n}+\epsilon_{n}
-regression xs ys = do
+regression :: Kernel [(Double, Double)] (Double, Double, Double)
+regression xsys = do
     params@(slope, intercept, noise) <- paramPriorRegression
```
```suggestion
    (slope, intercept, noise) <- paramPriorRegression
```
You do not need `params` name - it is not used anywhere here.

mknorps commented on docs/source/probprog.md:
```diff
@@ -18,39 +18,38 @@ y_{n}=\alpha+\beta x_{n}+\epsilon_{n}
 
 ![](_static/priorpred.png)
 
 We could then run the model as follows:
```
```suggestion
To run the model
```
Check out the video on how to write the documentation: https://www.youtube.com/watch?v=8TD-20Mb_7M.
It is a good compilation of advices :)

reubenharry replied:
Oh thanks for this, it looks super helpful!

mknorps commented on docs/source/probprog.md:
```diff
@@ -74,15 +73,22 @@ A distribution in monad-bayes over a set {math}`X`, is of type:
+
+which is distributed uniformly over {math}`[0,1]`.
 
 The full set is listed at https://hackage.haskell.org/package/monad-bayes-0.1.1.0/docs/Control-Monad-Bayes-Class.html
```
Can version of the documentation be taken as a variable? It would be a one less thing to remember when we bump the version.

mknorps replied:
Also, a link would be nice here.

reubenharry replied:
I agree, but don't know how.

mknorps replied:
Created an issue #203

mknorps reviewed (changes requested):
Hi Reuben!
If it is OK with you I will split the review in few parts, so we can have a discussion about each one of them.
First of all, I understood from the documentation and conversations with @idontgetoutmuch  what this package is about. In the next parts of review I will include some results of our conversations, that I think would make it easier for a newcomer to undestand the library.
For this chunk of review:
- please remove .ipynb_checkpoints from the PR
- given nice type synonyms `Distribution`, `Measure` and `Kernel` please use it instead of `MonadSample` and `MonadInfer` in the documentation - it will be easier to read and also gives a coherence

reubenharry committed:
Update README.md

Co-authored-by: Maria Knorps <maria.knorps@gmail.com>

reubenharry committed:
Update monad-bayes-site/about.md

Co-authored-by: Maria Knorps <maria.knorps@gmail.com>

reubenharry committed:
Update monad-bayes-site/about.md

Co-authored-by: Maria Knorps <maria.knorps@gmail.com>

reubenharry committed:
Update docs/source/probprog.md

Co-authored-by: Maria Knorps <maria.knorps@gmail.com>

reubenharry committed:
Update docs/source/probprog.md

Co-authored-by: Maria Knorps <maria.knorps@gmail.com>

reubenharry committed:
Update README.md

Co-authored-by: Arnaud Spiwack <arnaud@spiwack.net>

github-actions[bot] committed:
flake.lock: Update (#195)

Flake lock file updates:

• Updated input 'jupyter-flake':
    'git+https://github.com/tweag/monad-bayes?ref=notebooks&rev=a123b9352203dcd245bbd06a3139edb3c6f653ae' (2022-08-30)
  → 'git+https://github.com/tweag/monad-bayes?ref=notebooks&rev=4ef694502bf20a20790a0ad39a7db89750730dc2' (2022-09-23)
• Added input 'jupyter-flake/flake-compat':
    'github:edolstra/flake-compat/b4a34015c698c7793d592d66adbab377907a2be8' (2022-04-19)
• Added input 'jupyter-flake/haskell-nix-utils':
    'github:TerrorJack/haskell-nix-utils/f6fafb5f7ad7ab100bc897e9a278caabab38505d' (2022-07-12)
• Added input 'jupyter-flake/haskell-nix-utils/cabal-extras':
    'github:phadej/cabal-extras/7a8f07aafcdbbeaeefcbb5f7e8e0e12f91b59fcb' (2022-05-25)
• Added input 'jupyter-flake/haskell-nix-utils/cabal-fmt':
    'github:phadej/cabal-fmt/6651ffdccdfce71330f2b5cde9f8f23b616abf82' (2021-08-30)
• Added input 'jupyter-flake/haskell-nix-utils/ekg-json':
    'github:pepeiborra/ekg-json/7a0af7a8fd38045fd15fb13445bdcc7085325460' (2022-05-02)
• Added input 'jupyter-flake/haskell-nix-utils/eventlog2html':
    'github:mpickering/eventlog2html/e959fdf879c2b404a9843d962b726af499b27cb8' (2021-12-06)
• Added input 'jupyter-flake/haskell-nix-utils/gentle-introduction':
    'github:phadej/gentle-introduction/949a99b4d2d8c556bdd455f0e4c4d94f0402ea63' (2022-05-18)
• Added input 'jupyter-flake/haskell-nix-utils/ghcid':
    'github:ndmitchell/ghcid/f48626bf5b64067a84b644bcc56afac686046966' (2022-03-26)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-language-server':
    'github:haskell/haskell-language-server/445192e21daa4f4973e60a13e748ffa910819e79' (2022-07-11)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix':
    'github:input-output-hk/haskell.nix/66ae51e640c624519ba1bb1559a9ef2deccd9726' (2022-07-12)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/HTTP':
    'github:phadej/HTTP/9bc0996d412fef1787449d841277ef663ad9a915' (2016-01-01)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/cabal-32':
    'github:haskell/cabal/48bf10787e27364730dd37a42b603cee8d6af7ee' (2020-10-26)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/cabal-34':
    'github:haskell/cabal/942639c18c0cd8ec53e0a6f8d120091af35312cd' (2021-12-24)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/cabal-36':
    'github:haskell/cabal/f27667f8ec360c475027dcaee0138c937477b070' (2022-01-08)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/cardano-shell':
    'github:input-output-hk/cardano-shell/9392c75087cb9a3d453998f4230930dea3a95725' (2020-12-21)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/flake-utils':
    'github:numtide/flake-utils/3cecb5b042f7f209c56ffd8371b2711a290ec797' (2022-02-07)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/ghc-8.6.5-iohk':
    'github:input-output-hk/ghc/95713a6ecce4551240da7c96b6176f980af75cae' (2020-09-24)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/hackage':
    'github:input-output-hk/hackage.nix/f596b29505d4a1b5d561c7c270a9c3b147534468' (2022-07-12)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/hpc-coveralls':
    'github:sevanspowell/hpc-coveralls/14df0f7d229f4cd2e79f8eabb1a740097fdfa430' (2020-12-09)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/hydra':
    'github:NixOS/hydra/28b682b85b7efc5cf7974065792a1f22203a5927' (2022-03-10)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/hydra/nix':
    'github:NixOS/nix/a1cd7e58606a41fcf62bf8637804cf8306f17f62' (2022-01-24)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/hydra/nix/lowdown-src':
    'github:kristapsdz/lowdown/d2c2b44ff6c27b936ec27358a2653caaef8f73b8' (2021-10-06)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/hydra/nix/nixpkgs':
    'github:NixOS/nixpkgs/82891b5e2c2359d7e58d08849e4c89511ab94234' (2021-09-28)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/hydra/nix/nixpkgs-regression':
    'github:NixOS/nixpkgs/215d4d0fd80ca5163643b03a33fde804a29cc1e2' (2022-01-24)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/hydra/nixpkgs':
    follows 'jupyter-flake/haskell-nix-utils/haskell-nix/hydra/nix/nixpkgs'
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/nix-tools':
    'github:input-output-hk/nix-tools/e109c94016e3b6e0db7ed413c793e2d4bdb24aa7' (2022-04-08)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/nixpkgs':
    follows 'jupyter-flake/haskell-nix-utils/haskell-nix/nixpkgs-unstable'
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/nixpkgs-2003':
    'github:NixOS/nixpkgs/1db42b7fe3878f3f5f7a4f2dc210772fd080e205' (2021-05-03)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/nixpkgs-2105':
    'github:NixOS/nixpkgs/530a53dcbc9437363471167a5e4762c5fcfa34a1' (2022-02-19)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/nixpkgs-2111':
    'github:NixOS/nixpkgs/0a58eebd8ec65ffdef2ce9562784123a73922052' (2022-03-31)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/nixpkgs-unstable':
    'github:NixOS/nixpkgs/30d3d79b7d3607d56546dd2a6b49e156ba0ec634' (2022-03-25)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/old-ghc-nix':
    'github:angerman/old-ghc-nix/af48a7a7353e418119b6dfe3cd1463a657f342b8' (2021-09-08)
• Added input 'jupyter-flake/haskell-nix-utils/haskell-nix/stackage':
    'github:input-output-hk/stackage.nix/cfa19244bdb8ac3562780a0199fccae1a9220b43' (2022-07-12)
• Added input 'jupyter-flake/haskell-nix-utils/hindent':
    'github:mihaimaruseac/hindent/4c2ea034f4365cd784539f223282907c9e734fba' (2022-06-26)
• Added input 'jupyter-flake/haskell-nix-utils/hooglite':
    'github:phadej/hooglite/18856375932f6744cac7849bd1e816538537863f' (2021-12-29)
• Added input 'jupyter-flake/haskell-nix-utils/pointfree':
    'github:bmillwood/pointfree/2fb1b2e1a48b7297b4f157eb4f3085ad6fa40443' (2021-04-10)
• Added input 'jupyter-flake/haskell-nix-utils/warp-wo-x509':
    'github:phadej/warp-wo-x509/98648f7520d228e6a14747223f0bbd68620b9318' (2021-11-14)
• Added input 'jupyter-flake/haskell-nix-utils/weeder':
    'github:ocharles/weeder/c58ed2a8c66dcf0b469f8343efb6b6f61c7c40f3' (2022-05-22)
• Added input 'jupyter-flake/jupyter-flake':
    'git+https://github.com/tweag/monad-bayes?ref=notebooks&rev=a123b9352203dcd245bbd06a3139edb3c6f653ae' (2022-08-30)
• Added input 'jupyter-flake/jupyter-flake/flake-utils':
    'github:numtide/flake-utils/c0e246b9b83f637f4681389ecabcb2681b4f3af0' (2022-08-07)
• Added input 'jupyter-flake/jupyter-flake/jupyterWith':
    'github:tweag/jupyterWith/0b7f2e843f023c89283daf53eabce322efc9ca7c' (2022-08-29)
• Added input 'jupyter-flake/jupyter-flake/jupyterWith/flake-compat':
    'github:teto/flake-compat/8e15c6e3c0f15d0687a2ab6ae92cc7fab896bfed' (2021-09-15)
• Added input 'jupyter-flake/jupyter-flake/jupyterWith/flake-utils':
    'github:numtide/flake-utils/c0e246b9b83f637f4681389ecabcb2681b4f3af0' (2022-08-07)
• Added input 'jupyter-flake/jupyter-flake/jupyterWith/ihaskell':
    'github:gibiansky/IHaskell/53f8e0773822ddf2cd392309ba27c8ff8e30202e' (2022-08-26)
• Added input 'jupyter-flake/jupyter-flake/jupyterWith/ihaskell/flake-compat':
    'github:edolstra/flake-compat/12c64ca55c1014cdc1b16ed5a804aa8576601ff2' (2021-08-02)
• Added input 'jupyter-flake/jupyter-flake/jupyterWith/ihaskell/flake-utils':
    'github:numtide/flake-utils/7e5bf3925f6fbdfaf50a2a7ca0be2879c4261d19' (2021-09-13)
• Added input 'jupyter-flake/jupyter-flake/jupyterWith/ihaskell/hls':
    'github:haskell/haskell-language-server/311107eabbf0537e0c192b2c377d282505b4eff1' (2021-11-18)
• Added input 'jupyter-flake/jupyter-flake/jupyterWith/ihaskell/hls/flake-compat':
    'github:edolstra/flake-compat/12c64ca55c1014cdc1b16ed5a804aa8576601ff2' (2021-08-02)
• Added input 'jupyter-flake/jupyter-flake/jupyterWith/ihaskell/hls/flake-utils':
    'github:numtide/flake-utils/997f7efcb746a9c140ce1f13c72263189225f482' (2021-08-20)
• Added input 'jupyter-flake/jupyter-flake/jupyterWith/ihaskell/hls/gitignore':
    'github:hercules-ci/gitignore.nix/211907489e9f198594c0eb0ca9256a1949c9d412' (2021-01-26)
• Added input 'jupyter-flake/jupyter-flake/jupyterWith/ihaskell/hls/nixpkgs':
    'github:NixOS/nixpkgs/5e47a07e9f2d7ed999f2c7943b0896f5f7321ca3' (2021-09-06)
• Added input 'jupyter-flake/jupyter-flake/jupyterWith/ihaskell/hls/pre-commit-hooks':
    'github:cachix/pre-commit-hooks.nix/397f0713d007250a2c7a745e555fa16c5dc8cadb' (2021-06-29)
• Added input 'jupyter-flake/jupyter-flake/jupyterWith/ihaskell/hls/pre-commit-hooks/flake-utils':
    follows 'jupyter-flake/jupyter-flake/jupyterWith/ihaskell/hls/flake-utils'
• Added input 'jupyter-flake/jupyter-flake/jupyterWith/ihaskell/hls/pre-commit-hooks/nixpkgs':
    follows 'jupyter-flake/jupyter-flake/jupyterWith/ihaskell/hls/nixpkgs'
• Added input 'jupyter-flake/jupyter-flake/jupyterWith/ihaskell/nixpkgs':
    'github:NixOS/nixpkgs/5f0194220f2402b06f7f79bba6351895facb5acb' (2021-10-18)
• Added input 'jupyter-flake/jupyter-flake/jupyterWith/nixpkgs':
    'github:nixos/nixpkgs/eabc38219184cc3e04a974fe31857d8e0eac098d' (2022-08-02)
• Added input 'jupyter-flake/jupyter-flake/nixpkgs':
    'github:NixOS/nixpkgs/ce6aa13369b667ac2542593170993504932eb836' (2022-05-30)
• Removed input 'jupyter-flake/jupyterWith'
• Removed input 'jupyter-flake/jupyterWith/flake-compat'
• Removed input 'jupyter-flake/jupyterWith/flake-utils'
• Removed input 'jupyter-flake/jupyterWith/ihaskell'
• Removed input 'jupyter-flake/jupyterWith/ihaskell/flake-compat'
• Removed input 'jupyter-flake/jupyterWith/ihaskell/flake-utils'
• Removed input 'jupyter-flake/jupyterWith/ihaskell/hls'
• Removed input 'jupyter-flake/jupyterWith/ihaskell/hls/flake-compat'
• Removed input 'jupyter-flake/jupyterWith/ihaskell/hls/flake-utils'
• Removed input 'jupyter-flake/jupyterWith/ihaskell/hls/gitignore'
• Removed input 'jupyter-flake/jupyterWith/ihaskell/hls/nixpkgs'
• Removed input 'jupyter-flake/jupyterWith/ihaskell/hls/pre-commit-hooks'
• Removed input 'jupyter-flake/jupyterWith/ihaskell/hls/pre-commit-hooks/flake-utils'
• Removed input 'jupyter-flake/jupyterWith/ihaskell/hls/pre-commit-hooks/nixpkgs'
• Removed input 'jupyter-flake/jupyterWith/ihaskell/nixpkgs'
• Removed input 'jupyter-flake/jupyterWith/nixpkgs'
• Updated input 'jupyter-flake/nixpkgs':
    'github:NixOS/nixpkgs/ce6aa13369b667ac2542593170993504932eb836' (2022-05-30)
  → 'github:NixOS/nixpkgs/2da64a81275b68fdad38af669afeda43d401e94b' (2022-09-01)
• Added input 'jupyter-flake/pre-commit-hooks':
    'github:cachix/pre-commit-hooks.nix/8cb8ea5f1c7bc2984f460587fddd5f2e558f6eb8' (2022-08-18)
• Added input 'jupyter-flake/pre-commit-hooks/flake-utils':
    follows 'jupyter-flake/flake-utils'
• Added input 'jupyter-flake/pre-commit-hooks/nixpkgs':
    follows 'jupyter-flake/nixpkgs'
• Updated input 'nixpkgs':
    'github:NixOS/nixpkgs/2da64a81275b68fdad38af669afeda43d401e94b' (2022-09-01)
  → 'github:NixOS/nixpkgs/ae1dc133ea5f1538d035af41e5ddbc2ebcb67b90' (2022-09-22)
• Updated input 'pre-commit-hooks':
    'github:cachix/pre-commit-hooks.nix/8cb8ea5f1c7bc2984f460587fddd5f2e558f6eb8' (2022-08-18)
  → 'github:cachix/pre-commit-hooks.nix/60cad1a326df17a8c6cf2bb23436609fdd83024e' (2022-09-13)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

mknorps committed:
Remove notebook's checkpoints and add them to .gitignore (#200)

github-actions[bot] committed:
flake.lock: Update (#199)

Flake lock file updates:

• Updated input 'nixpkgs':
    'github:NixOS/nixpkgs/ae1dc133ea5f1538d035af41e5ddbc2ebcb67b90' (2022-09-22)
  → 'github:NixOS/nixpkgs/10ecda252ce1b3b1d6403caeadbcc8f30d5ab796' (2022-09-30)
• Updated input 'pre-commit-hooks':
    'github:cachix/pre-commit-hooks.nix/60cad1a326df17a8c6cf2bb23436609fdd83024e' (2022-09-13)
  → 'github:cachix/pre-commit-hooks.nix/2a4f1cfaa01b8b31edc7d3004454c4a0c38d50d8' (2022-09-30)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

mknorps committed:
Update monad-bayes-site/about.md

mknorps committed:
Maria/website adjustments (#201)

* Start working on site

* Fix formatting

* revert flake.nix

* clean up the PR files

* Do not change flake.lock

mknorps committed:
UPD readme with clearer instruction of notebooks run

mknorps committed:
Update nix flakes lock

mknorps committed:
UPD envrc (#202)

* UPD envrc

* Update .envrc

Co-authored-by: idontgetoutmuch <dominic@steinitz.org>

mknorps committed:
Notebooks: Introduction.ipynb adjustments (#205)

* Merge flakes

* The one flake to rule them all

* Fix plotCdf function & add central histogram point

* DEL Introduction notebook form notebooks lvl

UPD html rendering of the introduction notebook

* Update nix flakes lock

Co-authored-by: Dominic Steinitz <dominic@steinitz.org>

mknorps committed:
DEL duplicate notebooks in the  folder

mknorps commented on src/Control/Monad/Bayes/Sampler/Lazy.hs:
```diff
@@ -67,5 +71,44 @@ sampler m = newStdGen *> (runSampler m . randomTree <$> getStdGen)
+    bridge (Just (x, x')) y (Just (z, z')) = normal (x' + ((y - x) * (z' - x') / (z - x))) (sqrt ((z - y) * (y - x) / (z - x)))
+    bridge _ _ _ = error "undefined behavior for bridge"
+
+    findMaxLower :: Double -> [Double] -> Maybe Double
```
Alternative proposition of the implementation - may be easier to read:
```
findMaxLower :: Double -> [Double] -> Maybe Double
findMaxLower d = foldl' (upd d) Nothing
  where
    upd limit acc x = if x >= limit then acc
                      else max acc (Just x)
```

reubenharry replied:
That seems smart

mknorps replied:
not needed anymore

mknorps commented on src/Control/Monad/Bayes/Sampler/Lazy.hs:
```diff
@@ -67,5 +71,44 @@ sampler m = newStdGen *> (runSampler m . randomTree <$> getStdGen)
 weightedsamples :: Weighted Sampler a -> IO [(a, Log Double)]
 weightedsamples = sampler . independent . weighted
+
+wiener :: Sampler (Double -> Double)
```
@reubenharry Do you have some example of usage of the Wiener process generation?
It is a hidden gem now.

reubenharry replied:
There are examples in LazyPPL (source code on bitbucket) but perhaps it should be removed in Monad-Bayes.

mknorps replied:
resolved in #215

mknorps commented on src/Control/Monad/Bayes/Sampler/Lazy.hs:
```diff
@@ -67,5 +71,44 @@ sampler m = newStdGen *> (runSampler m . randomTree <$> getStdGen)
+
+wiener :: Sampler (Double -> Double)
+wiener = Sampler $ \(Tree _ gs) ->
+  unsafePerformIO $ do
```
Why we need to use `unsafePerformIO` here?

reubenharry replied:
The LazyPPL folks told me this was unavoidable for theoretical reasons. But as per above, I suggest we just remove the wiener process from the library.

mknorps replied:
resolved in #215

mknorps committed:
resolve warnings - shadowed variables and unused imports

mknorps committed:
DEL Wiener process (#215)

mknorps committed:
Merge branch 'master' into notebooks

mknorps committed:
Manual update of flake.lock
